packages/gatsby/src/bootstrap/index.js

```diff
@@ -75,6 +75,16 @@ module.exports = async (args: BootstrapArgs) => {
     getConfigFile(program.directory, `gatsby-config`)
   )
 
+  // The root config cannot be exported as a function, only theme configs
+  if (typeof config === `function`) {
+    report.panic({
+      id: `10126`,
+      context: {
+        configName: `gatsby-config`,
```

Comment:
This should pass in the path to the config file as well so it's clear which one to use.